test/test_postponed_annotations.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+import sys
+from pathlib import Path
+
+import pytest
+from dataclasses import dataclass
+from simple_parsing import ArgumentParser, Serializable
+
+from .b import B
+
+
+@pytest.mark.parametrize(
+    'sys_argv, b_value, p_value', [
+        (['test.py', '--v', '1'], 1, None),
+        (['test.py', '--v', '2', '--p', 'test/'], 2, Path('test/')),
+        (['test.py', '--v', '3', '--p', 'test/test1'], 3, Path('test/test1')),
+        pytest.param(['test.py'], None, None, marks=pytest.mark.xfail(reason='no default value in the dataclass'))
+    ]
+)
+def test_postponed_annotations_with_baseclass(sys_argv, b_value, p_value, monkeypatch):
```

Comment:
Nitpick: consider typing functions arguments
```suggestion
def test_postponed_annotations_with_baseclass(sys_argv: list[str], b_value: int | None, p_value: Path | None, monkeypatch):
```